app/services/InMemoryTrafficLightService.scala

```diff
@@ -27,23 +25,18 @@ class CachingTrafficLightService @Inject() extends TrafficLightService {
     db.get(id)
   }
 
-  override def getFuture(id: Int): Future[TrafficLight] = Future {
-    db(id)
+  override def getFuture(id: Int): Future[Option[TrafficLight]] = Future {
```

Comment:
Should be `Option`, because we cannot guarantee that the traffic light with that id exists.